### PR TITLE
Added ability to auto-resume from suspended state

### PR DESCRIPTION
**Pull request by simster7**
Closes: https://github.com/argoproj/argo/issues/1709. Adds field `.suspend.resumeAfterSeconds` to specify how much time to wait before auto-resuming a suspended state.

Example:
```
apiVersion: argoproj.io/v1alpha1
kind: Workflow
metadata:
  generateName: suspend-template-
spec:
  entrypoint: suspend
  templates:
  - name: suspend
    steps:
    - - name: approve
        template: approve
    - - name: release
        template: whalesay

  - name: approve
    suspend:
      resumeAfterSeconds: 5

  - name: whalesay
    container:
      image: docker/whalesay
      command: [cowsay]
      args: ["hello world"]
```

### CODE REVIEW

**Comment by simster7** [comment]
@sarabala1979 Changed the language to `resumeAfterSeconds`, added tests, and added examples. Should be ready for another review.

File: examples/suspend-template.yaml [hunk 1]
```diff
@@ -18,12 +21,18 @@ spec:
...
+        template: delay
     - - name: release
         template: whalesay
 
   - name: approve
     suspend: {}
 
+  - name: delay
+    suspend:
+      resumeAfterSeconds: 20
```
Comment (sarabala1979) [inline examples/suspend-template.yaml]: I think `duration` or `wakeup` will fit for this functionality. @jessesuen can comment

Comment (simster7) [reply examples/suspend-template.yaml]: `duration` is good. Maybe `durationSeconds`?

Comment (sarabala1979) [reply examples/suspend-template.yaml]: `duration` is good.

**Pull request merged**